src/reducers/chatReducer.js

```diff
@@ -26,7 +26,7 @@ const initialState = {
   previousUsername: previousUsername,
   status: '',
   messages: [],
-  statuses: [],
+  statuses: {},
```

Comment:
Is this correct?

I think I made it an array instead of an object to avoid the race condition where the client deletes a user's status right before that (still-connected) client sends a new status.

(...Not that those statuses are actually being deleted correctly, as you may have noticed; if someone ever connected during your current visit, the UI shows them as still being connected, even if that's not true. Maybe the big refactor I mentioned earlier broke those deletions...)

Reply:
Not sure if this is correct, but this is consistent with the object the app ends up creating for statuses.

I also expected it to be an array, but the actual variable being passed around was an object with username keys.

Reply:
Gotcha; makes sense.

Reply:
Checking out this branch (in both senses of the term) in a sec.